controllers/capabilities/product_controller.go

```diff
@@ -329,3 +362,30 @@ func computeBackendUsageList(list []capabilitiesv1beta1.Backend, backendUsageMap
 
 	return result
 }
+
+func (r *ProductReconciler) removeProduct(productResource *capabilitiesv1beta1.Product) error {
+	providerAccount, err := controllerhelper.LookupProviderAccount(r.Client(), productResource.Namespace, productResource.Spec.ProviderAccountRef, r.Logger())
+	if err != nil {
+		return err
+	}
+
+	threescaleAPIClient, err := controllerhelper.PortaClient(providerAccount)
+	if err != nil {
+		return err
+	}
+
+	err = threescaleAPIClient.DeleteProduct(*productResource.Status.ID)
+	if err != nil {
+		return err
+	}
+
+	//Confirm that product has been removed
+	products, err := threescaleAPIClient.ListProducts()
```

Comment:
no need to check. If the 3scale API does not return an error, the controller can assume it has been deleted.

Reply:
true, thanks Eguzki